package/ios/RNCSliderComponentView.mm

```diff
@@ -0,0 +1,253 @@
+            UIImage *minimumTrackImage = [trackImage resizableImageWithCapInsets:(UIEdgeInsets){
+                0, width, 0, width
+            } resizingMode:UIImageResizingModeStretch];
+            [slider setMinimumTrackImage:minimumTrackImage forState:UIControlStateNormal];
+        }
+        if (!_maximumTrackImageSet) {
+            UIImage *maximumTrackImage = [trackImage resizableImageWithCapInsets:(UIEdgeInsets){
+                0, width, 0, width
+            } resizingMode:UIImageResizingModeStretch];
+            [slider setMaximumTrackImage:maximumTrackImage forState:UIControlStateNormal];
+        }
+    }
+}
+
+- (void)setMinimumTrackImage:(UIImage *)minimumTrackImage
+{
+    _trackImage = nil;
+    _minimumTrackImageSet = true;
+    minimumTrackImage = [minimumTrackImage resizableImageWithCapInsets:(UIEdgeInsets){
+        0, minimumTrackImage.size.width, 0, 0
+    } resizingMode:UIImageResizingModeStretch];
+    [slider setMinimumTrackImage:minimumTrackImage forState:UIControlStateNormal];
+}
+
+- (UIImage *)minimumTrackImage
+{
+    return [slider thumbImageForState:UIControlStateNormal];
+}
+
+- (void)setMaximumTrackImage:(UIImage *)maximumTrackImage
+{
+    _trackImage = nil;
+    _maximumTrackImageSet = true;
+    maximumTrackImage = [maximumTrackImage resizableImageWithCapInsets:(UIEdgeInsets){
+        0, 0, 0, maximumTrackImage.size.width
+    } resizingMode:UIImageResizingModeStretch];
+    [slider setMaximumTrackImage:maximumTrackImage forState:UIControlStateNormal];
+}
+
+- (UIImage *)maximumTrackImage
+{
+    return [slider thumbImageForState:UIControlStateNormal];
+}
+
+- (void)setThumbImage:(UIImage *)thumbImage
+{
+    [slider setThumbImage:thumbImage forState:UIControlStateNormal];
+}
+
+
+- (UIImage *)thumbImage
+{
+    return [slider thumbImageForState:UIControlStateNormal];
+}
+
+- (void)setInverted:(BOOL)inverted
+{
+    if (inverted) {
+        slider.transform = CGAffineTransformMakeScale(-1, 1);
+    } else {
+        slider.transform = CGAffineTransformMakeScale(1, 1);
+    }
+}
+
+- (float)discreteValue:(float)value
+{
+    if (self.step > 0 && value >= slider.maximumValue) {
+        return slider.maximumValue;
+    }
+    
+    if (self.step > 0 && self.step <= (slider.maximumValue - slider.minimumValue)) {
+        double (^_round)(double) = ^(double x) {
+            if (!UIAccessibilityIsVoiceOverRunning()) {
+                return round(x);
+            } else if (self.lastValue > value) {
+                return floor(x);
+            } else {
+                return ceil(x);
+            }
+        };
+        
+        return MAX(slider.minimumValue,
+                   MIN(slider.maximumValue, slider.minimumValue + _round((value - slider.minimumValue) / self.step) * self.step)
+                   );
+    }
+    
+    return value;
+}
+
+@end
+
+Class<RCTComponentViewProtocol> RNCSliderCls(void)
+{
+    return RNCSliderComponentView.class;
+}
+
+#endif // RCT_NEW_ARCH_ENABLED
```

Comment:
Should I get rid of the new arch if or only the comment? Because I'm checking for the new arch in the two files that I've added so I'm consistent with it